Tools/autotest/arducopter.py

```diff
@@ -1129,6 +1132,89 @@ def fly_mission(self, height_accuracy=-1.0, target_altitude=None):
         self.mavproxy.send('switch 5\n')  # loiter mode
         self.wait_mode('LOITER')
 
+    def fly_vision_position(self):
+        '''disable GPS navigation, enable Vicon input'''
+        # scribble down a location we can set origin to:
+
+        self.progress("Waiting for location")
+        start = self.mav.location()
+        self.mavproxy.send('switch 6\n')  # stabilize mode
+        self.mav.wait_heartbeat()
+        self.wait_mode('STABILIZE')
+        self.progress("Waiting reading for arm")
+        self.wait_ready_to_arm()
+
+        old_pos = self.mav.recv_match(type='GLOBAL_POSITION_INT', blocking=True)
+        print("old_pos=%s" % str(old_pos))
+
+        old_gps_type = self.get_parameter("GPS_TYPE")
+        old_ek2_gps_type = self.get_parameter("EK2_GPS_TYPE")
```

Comment:
I think it would more interesting to test SIM_GPS_TYPE set to 0 (or even SIM_GPS_DISABLE set to 1, but I'm not sure that works).

Reply:
That would be an interesting test - but you'd essentially be testing a hardware failure condition rather than a machine configuration.

*I* think an interesting configuration would be to leave the GPS enabled - do the two systems play nice?

Reply:
> but you'd essentially be testing a hardware failure condition rather than a machine configuration.

@peterbarker Quite the contrary, SIM_GPS_TYPE set to 0 means an hardware configuration where a GPS isn't connected. The ones you used are forcing a connected GPS to not be used.

> I think an interesting configuration would be to leave the GPS enabled - do the two systems play nice?

Yes, that would also be an interesting test.